_posts/2023-04-21-extensibility-roadmap.markdown

```diff
@@ -0,0 +1,154 @@
+
+We will similarly support extensions in the clients, and creating thin-clients per extension which will be composable with the core client in every supported programming language. Extensions will publish REST interfaces in the shape of OpenAPI specs, a generator will consume the spec and output a complete high level thin client. The burden of m * n extensions and languages will be alleviated by automating as much of the process as possible, providing build and test tooling such as CI workflows, so that both project owned extensions, and third party developed extensions alike, can benefit with uniform support. The extension owner can then take the generated clients and publish them to their package repositories of choice as desired. The core clients will define stable low-level/raw interface with its transport layer such that the thin-clients compose as expected and follow semver compatibility rules.
+
+### Rewrite most plugins as extensions
+
+We will provide a line of sight for deprecating plugins by enabling extension migration with as close to one-to-one correspondence with plugin features as possible with the aim of making the effort required to migrate to extensions minimal. We will write a migration guide clearly specifying the needed effort to perform initial migration and follow-up deprecated feature replacement so that they can work it into their own task planning, offer code that makes the migration from plugins to extensions easier, implement samples that provide one-to-one analogues with the existing plugin framework, and create assurances that external behavior in the migration has not changed. Code permitting this quick bridge may be marked deprecated, but will allow users to methodically remove the deprecated code over time.
+
+This is [opensearch-sdk-java#315](https://github.com/opensearch-project/opensearch-sdk-java/issues/315).
+
+### Deprecate plugins, unbundle distributions
+
+Assuming extensions have been widely adopted, we can deprecate the plugin APIs to be removed in the next major version of OpenSearch. We don't expect this to happen earlier than OpenSearch 4.0. Older versions of plugins continue to work with older versions of OpenSearch and continue receiving security patches.
+
+We will replace the two distribution flavors of OpenSearch/Dashboards (currently a -min distribution without security or plugins) by a set of distribution manifests tailored for purposes such as log analytics vs. search. Each distribution will be represented by a manifest that can be assembled “on-the-fly” by downloading the artifacts from the public catalog for packaging purposes. We want to enable single-click install a collection of extensions, recommended distributions of OpenSearch for tailored purposes, ability for vendors to create their favorite flavor of OpenSearch/Dashboards distribution easily, and add the capability to create enterprise-tailored distributions.
```

Comment:
If your intention is to key off of "enable", consider these changes...

"We want to enable single-click **installation of a collection** of extensions, recommended distributions of OpenSearch for tailored purposes, **the** ability for vendors to create their favorite flavor of OpenSearch/Dashboards distribution easily, **and the capability** to create enterprise-tailored distributions."